app/services/katello/pulp3/repository/apt.rb

```diff
@@ -0,0 +1,79 @@
+require 'pulp_deb_client'
+
+module Katello
+  module Pulp3
+    class Repository
+      class Apt < ::Katello::Pulp3::Repository
+        def remote_options
+          deb_remote_options = {
+            distributions: root.deb_releases
+          }
+          deb_remote_options[:components] = root.deb_components.present? ? root.deb_components : nil
+          deb_remote_options[:architectures] = root.deb_architectures.present? ? root.deb_architectures : nil
+
+          if root.url.blank?
+            deb_remote_options[:url] = nil
+          end
+
+          deb_remote_options[:gpgkey] = root.gpg_key.present? ? root.gpg_key.content : nil
+
+          common_remote_options.merge(deb_remote_options)
+        end
+
+        def publication_options(repository_version)
+          ss = api.signing_services_api.list(name: 'katello_deb_sign').results
```

Comment:
is the idea that the user needs to create this manually with this name?  Could we move this name to a constant?

Reply:
> is the idea that the user needs to create this manually with this name?

Yes, for now I intended it to be created from outside, because on pulp2 the process was also that the key and the signing-script had to be defined manually.

> Could we move this name to a constant?

You mean a constant within this class or a (foreman-)setting?

Reply:
in the class is fine, just so its not a 'magic string'